flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -488,16 +489,18 @@ private static PartitionSpec toPartitionSpec(List<String> partitionKeys, Schema
 
   private static List<String> toPartitionKeys(PartitionSpec spec, Schema icebergSchema) {
     List<String> partitionKeys = Lists.newArrayList();
+    ImmutableList.Builder<String> partitionKeysBuilder = ImmutableList.builder();
     for (PartitionField field : spec.fields()) {
       if (field.transform().isIdentity()) {
-        partitionKeys.add(icebergSchema.findColumnName(field.sourceId()));
+        partitionKeysBuilder.add(icebergSchema.findColumnName(field.sourceId()));
       } else {
         // Not created by Flink SQL.
         // For compatibility with iceberg tables, return empty.
         // TODO modify this after Flink support partition transform.
-        return Collections.emptyList();
+        return partitionKeys;
       }
     }
+    partitionKeys = partitionKeysBuilder.build();
     return partitionKeys;
```

Comment:
This could return partitionKeysBuilder.build()

Reply:
fixed